nati_zynq_defconfig: Cherry-pick all nati_zynq_defconfig related commits

DeborahOoi96 opened the pull request:
We need to update linux 6.6 kernel to support bluefin devices. Cherry-picking all nati_zynq_defconfig related commits from bluefin/master/4.1 into nilrt/master/6.6. This file is needed in order for linux-bluefin kernel recipe to build correctly.

WI: [AB#2959877](https://dev.azure.com/ni/DevCentral/_workitems/edit/2959877)

Testing
Unable to build kernel until all necessary commits are cherry-picked. nati_zynq_defconfig is only part of the changes needed to build.

Commit:
nati_zynq_defconfig: add initial defconfig for NI Zynq-based targets

Signed-off-by: Josh Cartwright <joshc@ni.com>
(cherry picked from commit fd9ef11f38fb2450b3d8f8849a673a97dff17252)
Signed-off-by: deooi <deborah.ooi@ni.com>

Commit:
nati_zynq_defconfig: disable PL310_ERRATA_588369

The PL310 errata 588369 is fixed in revision r2p0.

The Zynq chips we are using are documented as having revision
r3p2-50rel0 of the PL310, so this errata is not relevant on our
platforms.  Disable it.

Suggested-by: Gratian Crisan <gratian.crisan@ni.com>
Signed-off-by: Josh Cartwright <joshc@ni.com>
(cherry picked from commit 9e2a5f4f128bc13c34f273cac6331e8aa81c64b2)
Signed-off-by: deooi <deborah.ooi@ni.com>

Commit:
nati_zynq_defconfig: disable PL310_ERRATA_727915

The PL310 errata 727915 is fixed in revision r3p1.

The Zynq chips we are using are documented as having revision
r3p2-50rel0 of the PL310, so this errata is not relevant on our
platforms.  Disable it.

Suggested-by: Gratian Crisan <gratian.crisan@ni.com>
Signed-off-by: Josh Cartwright <joshc@ni.com>
(cherry picked from commit e72e0bfeed97dce479913dde59213374602f888e)
Signed-off-by: deooi <deborah.ooi@ni.com>

Commit:
nati_zynq_defconfig: Remove USB_OTG

The USB gadget support isn't dependent on USB_OTG support, hence we don't
need it in our defconfig.

Signed-off-by: Rishika Gupta <rishika.gupta@ni.com>
Signed-off-by: Brad Mouring <brad.mouring@ni.com>
Acked-by: Josh Cartwright <josh.cartwright@ni.com>
Acked-by: Gratian Crisan <gratian.crisan@ni.com>
Acked-by: Xander Huff <xander.huff@ni.com>
Natinst-ReviewBoard-ID: 118133
(cherry picked from commit a59ee85a2451df5019dfa880351c3f584ba52c48)
Signed-off-by: deooi <deborah.ooi@ni.com>

natesully committed:
nati_zynq_defconfig: Enable legacy irqdomain on the GIC

Signed-off-by: Nathan Sullivan <nathan.sullivan@ni.com>
Signed-off-by: Brad Mouring <brad.mouring@ni.com>
Acked-by: Xander Huff <xander.huff@ni.com>
Acked-by: Gratian Crisan <gratian.crisan@ni.com>

Natinst-CAR-ID: 567714
Natinst-ReviewBoard-ID: 123493
(cherry picked from commit 171a38afe3c1afedcf788967ecbc0ffcffed29b4)
Signed-off-by: deooi <deborah.ooi@ni.com>

Commit:
nati_zynq_defconfig: Regen to fix ordering issue

Some of the changes to the defconfig were cherry-picked from
different trees, resulting in option ordering issues. Regen to
fix this.

Signed-off-by: Brad Mouring <brad.mouring@ni.com>
(cherry picked from commit bdf4edbcecd280576730f6e9648ab1c99cb90be8)
Signed-off-by: deooi <deborah.ooi@ni.com>

natesully committed:
nati_zynq_defconfig: Enable GPIO LEDs

RoboRIO now uses GPIO LEDs for the radio lights.

Signed-off-by: Nathan Sullivan <nathan.sullivan@ni.com>
Signed-off-by: Brad Mouring <brad.mouring@ni.com>
Acked-by: Joseph Hershberger <joseph.hershberger@ni.com>

Natinst-CAR-ID: 563167
Natinst-ReviewBoard-ID: 126991
(cherry picked from commit 26a546c3e34d8a038ea9a46e6523f40da446f73d)
Signed-off-by: deooi <deborah.ooi@ni.com>

katieensign committed:
nati_zynq_defconfig: Enable CONFIG_USB_VIDEO_CLASS

The uvcvideo module is required for webcam support, and was
inadvertently removed when upgrading to the 4.1 kernel.

Signed-off-by: Katie Ensign <katie.ensign@ni.com>

Natinst-ReviewBoard-ID: 138583
Natinst-CAR-ID: 587538
(cherry picked from commit 684dcb123840e7faef2e22587caef475e63d0152)
Signed-off-by: deooi <deborah.ooi@ni.com>

(cherry picked from commit 642eab8dedb953aff0257e8e0c15cfc142686d34)
Signed-off-by: deooi <deborah.ooi@ni.com>

gratian commented:
A couple of notes:

- Please combine the fixup commit "[fixup! nati_zynq_defconfig: Enable CONFIG_USB_VIDEO_CLASS](https://github.com/ni/linux/pull/187/commits/0454fa584157f3d7d65cf15e084fb41ff8701436)" with the commit it fixes https://github.com/ni/linux/pull/187/commits/fd951ec47ef2bb166e2c73e5f674932be2923217 (i.e. squash the fixup)

- Like in PR https://github.com/ni/linux/pull/186 can you add a commit that does a round-trip through savedefconfig so it gets cleanly regenerated for 6.6

Commit:
nati_zynq_defconfig: round-trip through savedefconfig
Round-trip the configuration through savedefconfig so that it's ordered.

No functional changes to configuration otherwise.

Signed-off-by: deooi <deborah.ooi@ni.com>

gratian commented:
PR completed outside GitHub UI in order to correct a small issue with the commit message (no empty line after subject) for https://github.com/ni/linux/commit/df1771f9b9564f9eca2b60cd053f861476b9aacf ("nati_zynq_defconfig: round-trip through savedefconfig")
